### ISSUE TITLE
Application was unable to start correctly (0xc0000142)

### ISSUE THREAD

**Issue by oleteacher** [jyapayne/Web2Executable#207]
Thought I had awoke to a wonderful day when I saw "new release v0.7.0b":) Only for MAC...

We have not been able to use Web2Exe at all this year (or most of last year) in classroom projects. Running Win 10 and all computers attempting to run a compiled software from Web2Exe show:

![image](https://user-images.githubusercontent.com/26278259/64865002-f8da2980-d5fd-11e9-9009-c5bf1d7a9118.png)

Does not matter what we compile, even simple HTML only page produces above error. Error appears right after clicking on .exe - Have tried about everything found to correct issue, but no luck so thinking it must be Web2Exe?

Anyone else experience this issue and found a solution? We would love to start using Web2Exe again in classroom:)

Just so clear, Web2Exe starts and runs fine, it is the complied software that has issues.

**Comment by jyapayne** [comment]
@oleteacher I will be getting a release out for the other platforms soon (trying for today). I'm not certain why the builds using the current Web2Exe are failing, but I will look into it as soon as I get the Windows build up and running. Thanks for letting me know the issue!

**Comment by oleteacher** [comment]
Joey! So nice to hear from you, hope life is treating you well.

Good to know you are still trying to work on project, totally understand time restraints. I will tell you, students think it is the coolest thing to turn HTML into software - so you are appreciated:)

**Comment by jyapayne** [comment]
:) Life is treating me well, thank you for asking. And thank you for the kind words! Glad to know the kids enjoy turning HTML into software :D

I want to at least get the project in a working state for the people who still want to use it. I wish I could have set up a forum or something to track requests and to ask you guys for ideas. It's hard to know exactly how many people use this and it would have been nice to see and engage a little more with the community. I suppose it's never too late to start one.

**Comment by oleteacher** [comment]
Nice to hear everything is going well for you.

Sometimes outside of school some of us teachers need a simple forum to exchange ideas about things / projects. One of the boards we use is https://proboards.com/free-forum-features

Only drawback of that one is advertisements that you cannot control. But, it is hosted and you do not have to worry about maintaining yet another site/project. Might be something to look into. Sure there are many free ones but that is the only one we have ever used.

**Comment by jyapayne** [comment]
Thank you for the link. I'll check it out :)

Also, the new download is up for windows. Please check it out and see if your exported exe's work properly. I tried to fix it and it works on my computer, but that's never a good indication if it will work for everyone else.

**Comment by oleteacher** [comment]
Thanks Joey:)

I did a quick test to see how things are working:

- Creating/compiling a 32bit version on 64bit Win 10 machine: Results in same error message as show when trying to run 32bit exe

- Creating/compiling 64bit version on 64bit Win 10 machine: Runs perfectly!

I am not sure can get to a 32bit machine to test further. In the past we always created 32bit so would run on all machines. Honestly not even sure 32bit is still important these days, but sure they are still a many?

I will do some more testing over the coming week and report back. Hopefully others will see this and chime in if they are having same issue?

Really appreciate you taking time to get working again Joey!

**Comment by jyapayne** [comment]
Awesome, thanks for testing it out. Would you be able to provide the project files that create the error on the 32 bit build?

**Comment by oleteacher** [comment]
"Would you be able to provide the project files that create the error on the 32 bit build?"

Sure can, but honestly just a simple HTML website with some jquery and such. Just picked an old test website at random:) Let me round them up. Where shall I send them?

**Comment by jyapayne** [comment]
You should be able to just zip it up and drag/drop it to the text box here.

**Comment by oleteacher** [comment]
Ok, zipped it up. Hope it helps but really just a template that was generated by a builder.

[5d69f486c20cd_website.zip](https://github.com/jyapayne/Web2Executable/files/3613055/5d69f486c20cd_website.zip)

**Comment by oleteacher** [comment]
Quick question on another subject (do not wish to dirty this thread, but not worthy to open new): I see now that a folder called "package.nw" and contains all the source files. This exposes everything. How do I prevent this and all all files complied into exe? Sorry, been a while since used web2exe and do not recall this happening:)

**Comment by jyapayne** [comment]
Hmm, I can't seem to reproduce your error. Both the 32 bit and 64 bit versions work correctly for me. Perhaps it's a codec issue since you are playing an mp4 file. If you download the windows 32 bit zip file from [here](https://github.com/iteufel/nwjs-ffmpeg-prebuilt/releases) and drop it into the exported project (it should overwrite the ffmpeg.dll file there), does that help at all?

> I see now that a folder called "package.nw" and contains all the source files. This exposes everything. How do I prevent this and have all files complied into exe? Sorry, been a while since used web2exe and do not recall this happening:)

You can simply uncheck the "Uncompressed folder" option in the "Compression Settings" tab. However, Web2Exe does not hide the sources or obfuscate them in any way. For that you'd have to use something like [this](http://docs.nwjs.io/en/latest/For%20Users/Advanced/Protect%20JavaScript%20Source%20Code/) or some other tool.

**Comment by jyapayne** [comment]
Another thing to try is deleting the package.json from your project and filling out the fields you want again. There's a small chance that old settings might interfere with new ones.

**Comment by oleteacher** [comment]
Error must be only on my school computers then. Maybe something on them causing issue. Not the mp4 file, tried with another simple HTML site and it also produced the error on 32bit. Will keep searching for a solution but not much hope. May end with a 64bit only now that can produce those again:)

Forgot a bout the uncompressed folder solution. Not try to deeply hide, just did not want the files right in front of eyes.

**Comment by jyapayne** [comment]
> Error must be only on my school computers then. Maybe something on them causing issue. Not the mp4 file, tried with another simple HTML site and it also produced the error on 32bit. Will keep searching for a solution but not much hope. May end with a 64bit only now that can produce those again:)

What do your school computers run? Windows 10?

**Comment by oleteacher** [comment]
Yes, all computers are Win 10. They all are of course controlled by administrator (IT Department).

Broke my ole trusty personal laptop which runs Win 10 Pro and it to show the error when trying to run exe. Really strange. I hate not being able to fix something, but this one has me stumped. Have followed about every online tutorial on how to fix earlier this year and nothing solved it.

**Comment by oleteacher** [comment]
Hey Joey. What version of Windows are you running? I assume it is also 64bit?

Trying to troubleshoot and wondering your version.

**Comment by jyapayne** [comment]
Yep, I'm running Windows 10 64 bit. If it's not running on your system, it could be an issue with the NW.js version you're using as well. Although I assume you're using the latest?

**Comment by oleteacher** [comment]
Ok, got ya. Just to be clear we are still talking about compiled 32bit EXE produces "Application was unable to start correctly (0xc0000142)"

Everything else working fine. Yes, using the latest NW.js (do not have web2exe open, but think .41).

**Comment by PixlRainbow** [comment]
I'm having the same issue when building 32 bit Windows exe from a 64 bit Ubuntu 18.04 host. Running the built 32 bit exe on a 64 bit Windows 10 computer shows the same error code. I'm using v0.7.1b.

**Comment by jyapayne** [comment]
@PixlRainbow thanks for the info. I will try your specific configuration and get back to you

**Comment by jyapayne** [comment]
Hmm, I still can't reproduce. I created a 32 bit Windows exe using Ubuntu 19.04 (which should be the same). Running it on a 64 bit Windows installation works perfectly.

**Comment by PixlRainbow** [comment]
I wonder if the exe default attempts to use some globally installed external library, if present. And if it tries to use the 64 bit version if it can't find a 32 bit version.
You might have some extra libraries installed (so the external global libraries load correctly), or you might have a clean OS install (so NW.js uses the packaged libraries).

**Comment by 2braincells2go** [comment]
Can confirm what others are experiencing is exact experience I have. Can take a simple html file that has h1 hello /h1 and error happens on 32bit. 64bit runs without issue.
